### PR TITLE
Take a fee for prize messages and send to treasury

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/libraries/utils/src/consts.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ pub const CYCLES_REQUIRED_FOR_UPGRADE: Cycles = 300_000_000_000; // 0.3T cycles
 pub const OPENCHAT_BOT_USER_ID: UserId = UserId::new(Principal::from_slice(&[228, 104, 142, 9, 133, 211, 135, 217, 129, 1]));
 pub const DELETED_USER_ID: UserId = UserId::new(Principal::from_slice(&[139, 36, 200, 58, 72, 145, 241, 66, 97, 1]));
 pub const OPENCHAT_BOT_USERNAME: &str = "OpenChatBot";
+pub const OPENCHAT_TREASURY_CANISTER_ID: CanisterId = Principal::from_slice(&[0, 0, 0, 0, 2, 0, 0, 23, 1, 1]);
```
Comment (megrogan) [inline backend/libraries/utils/src/consts.rs:21]: This needs the real canister id

File: backend/canisters/community/impl/src/lib.rs [hunk 2]
```diff
@@ -240,7 +240,7 @@ impl RuntimeState {
                     next_event_expiry = Some(expiry);
                 }
             }
-            prize_refunds.extend(result.prize_refunds);
+            prize_refunds.extend(result.final_prize_payments);
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/lib.rs:243]: This should probably be named something like `payments_to_make` (feel free to come up with something better)

File: backend/canisters/group/impl/src/timer_job_types.rs [hunk 3]
```diff
@@ -235,14 +232,15 @@ impl Job for EndPollJob {
 
 impl Job for RefundPrizeJob {
```
Comment (hpeebles) [inline backend/canisters/group/impl/src/timer_job_types.rs:233]: `RefundPrizeJob` should be renamed too.

File: backend/canisters/community/impl/src/timer_job_types.rs [hunk 4]
```diff
@@ -22,7 +21,7 @@ pub enum TimerJob {
     FinalizeGroupImport(FinalizeGroupImportJob),
     ProcessGroupImportChannelMembers(ProcessGroupImportChannelMembersJob),
     MarkGroupImportComplete(MarkGroupImportCompleteJob),
-    RefundPrize(RefundPrizeJob),
+    FinalPrizePayments(FinalPrizePaymentsJob),
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/timer_job_types.rs:24]: This needs a serde alias to work from the old name

File: backend/libraries/chat_events/src/chat_events.rs [hunk 5]
```diff
@@ -1862,9 +1861,9 @@ impl ChatEvents {
                         });
                     }
                     if let MessageContentInternal::Prize(mut p) = m.content {
-                        if let Some(refund) = p.prize_refund(m.sender, &MEMO_PRIZE_REFUND, now * 1_000_000) {
-                            result.prize_refunds.push(refund);
-                        }
+                        result
+                            .final_prize_payments
+                            .append(&mut p.final_payments(m.sender, now * 1_000_000));
```
Comment (hpeebles) [inline backend/libraries/chat_events/src/chat_events.rs:1866]: I think `.extend(p.final_payments(m.sender, now * 1_000_000));` would be the more idiomatic way of doing this.

**Pull request merged**